[RDY]vim-patch:8.0.0451

janlazo opened the pull request:
**vim-patch:8.0.0451: some macros are in lower case**

Problem:    Some macros are in lower case.
Solution:   Make a few more macros upper case. Avoid lower case macros use an
            argument twice.
https://github.com/vim/vim/commit/91acfffc1e6c0d8c2abfb186a0e79a5bf19c3f3f

Related: https://github.com/neovim/neovim/issues/6297

ckelsel commented:
see also #6297

ckelsel commented:
It's better to use inline function

```
src/nvim/edit.c:2012:  Do not use preincrement in statements, use postincrement instead  [readability/increment] [5]
src/nvim/edit.c:2023:  Do not use preincrement in statements, use postincrement instead  [readability/increment] [5]
src/nvim/edit.c:3443:  for should always use braces  [readability/braces] [5]
src/nvim/edit.c:4601:  while should always use braces  [readability/braces] [5]
src/nvim/edit.c:4603:  if should always use braces  [readability/braces] [5]
src/nvim/eval.c:2585:  if should always use braces  [readability/braces] [5]
src/nvim/eval.c:18069:  /*-style comment found, it should be replaced with //-style.  /*-style comments are only allowed inside macros.  Note that you should not use /*-style comments to document macros itself, use doxygen-style comments for this.  [readability/old_style_comment] [5]
src/nvim/eval.c:18070:  for should always use braces  [readability/braces] [5]
src/nvim/eval.c:18071:  Line contains only semicolon. If this should be an empty statement, use {} instead.  [whitespace/semicolon] [5]
src/nvim/eval.c:22154:  /*-style comment found, it should be replaced with //-style.  /*-style comments are only allowed inside macros.  Note that you should not use /*-style comments to document macros itself, use doxygen-style comments for this.  [readability/old_style_comment] [5]
src/nvim/eval.c:22156:  if should always use braces  [readability/braces] [5]
src/nvim/eval.c:22242:  while should always use braces  [readability/braces] [5]
src/nvim/eval.c:22251:  while should always use braces  [readability/braces] [5]
src/nvim/ex_cmds.c:3273:  Do not use preincrement in statements, use postincrement instead  [readability/increment] [5]
src/nvim/ex_docmd.c:2894:  /*-style comment found, it should be replaced with //-style.  /*-style comments are only allowed inside macros.  Note that you should not use /*-style comments to document macros itself, use doxygen-style comments for this.  [readability/old_style_comment] [5]
src/nvim/ex_docmd.c:2894:  Do not use preincrement in statements, use postincrement instead  [readability/increment] [5]
src/nvim/ex_docmd.c:3343:  /*-style comment found, it should be replaced with //-style.  /*-style comments are only allowed inside macros.  Note that you should not use /*-style comments to document macros itself, use doxygen-style comments for this.  [readability/old_style_comment] [5]
src/nvim/ex_docmd.c:3343:  Do not use preincrement in statements, use postincrement instead  [readability/increment] [5]
src/nvim/ex_docmd.c:9489:  /*-style comment found, it should be replaced with //-style.  /*-style comments are only allowed inside macros.  Note that you should not use /*-style comments to document macros itself, use doxygen-style comments for this.  [readability/old_style_comment] [5]
src/nvim/ex_docmd.c:9490:  for should always use braces  [readability/braces] [5]
src/nvim/ex_docmd.c:9491:  if should always use braces  [readability/braces] [5]
src/nvim/fold.c:3059:  If an else has a brace on one side, it should have it on both  [readability/braces] [5]
src/nvim/mbyte.c:1746:  if should always use braces  [readability/braces] [5]
src/nvim/menu.c:1552:  for should always use braces  [readability/braces] [5]
src/nvim/menu.c:1553:  if should always use braces  [readability/braces] [5]
src/nvim/ops.c:409:  if should always use braces  [readability/braces] [5]
src/nvim/option.c:6213:  for should always use braces  [readability/braces] [5]
src/nvim/option.c:6214:  if should always use braces  [readability/braces] [5]
src/nvim/option.c:6388:  if should always use braces  [readability/braces] [5]
src/nvim/option.c:6431:  /*-style comment found, it should be replaced with //-style.  /*-style comments are only allowed inside macros.  Note that you should not use /*-style comments to document macros itself, use doxygen-style comments for this.  [readability/old_style_comment] [5]
src/nvim/path.c:864:  while should always use braces  [readability/braces] [5]
src/nvim/path.c:1089:  if should always use braces  [readability/braces] [5]
src/nvim/path.c:1448:  while should always use braces  [readability/braces] [5]
src/nvim/path.c:1450:  An else should always have braces before it  [readability/braces] [5]
src/nvim/path.c:1458:  while should always use braces  [readability/braces] [5]
src/nvim/path.c:1480:  while should always use braces  [readability/braces] [5]
src/nvim/regexp.c:1149:  if should always use braces  [readability/braces] [5]
src/nvim/regexp.c:1151:  If an else has a brace on one side, it should have it on both  [readability/braces] [5]
src/nvim/regexp.c:1189:  /*-style comment found, it should be replaced with //-style.  /*-style comments are only allowed inside macros.  Note that you should not use /*-style comments to document macros itself, use doxygen-style comments for this.  [readability/old_style_comment] [5]
src/nvim/regexp.c:1189:  if should always use braces  [readability/braces] [5]
src/nvim/regexp.c:4964:  An else should always have braces before it  [readability/braces] [5]
src/nvim/regexp.c:4964:  If an else has a brace on one side, it should have it on both  [readability/braces] [5]
src/nvim/regexp.c:4966:  if should always use braces  [readability/braces] [5]
src/nvim/regexp.c:5037:  If an else has a brace on one side, it should have it on both  [readability/braces] [5]
src/nvim/regexp.c:5168:  Do not use preincrement in statements, use postincrement instead  [readability/increment] [5]
src/nvim/regexp_nfa.c:1570:  Use true instead of TRUE.  [readability/bool] [4]
src/nvim/regexp_nfa.c:1670:  /*-style comment found, it should be replaced with //-style.  /*-style comments are only allowed inside macros.  Note that you should not use /*-style comments to document macros itself, use doxygen-style comments for this.  [readability/old_style_comment] [5]
src/nvim/regexp_nfa.c:1698:  Use true instead of TRUE.  [readability/bool] [4]
src/nvim/regexp_nfa.c:1772:  /*-style comment found, it should be replaced with //-style.  /*-style comments are only allowed inside macros.  Note that you should not use /*-style comments to document macros itself, use doxygen-style comments for this.  [readability/old_style_comment] [5]
src/nvim/regexp_nfa.c:1775:  /*-style comment found, it should be replaced with //-style.  /*-style comments are only allowed inside macros.  Note that you should not use /*-style comments to document macros itself, use doxygen-style comments for this.  [readability/old_style_comment] [5]
src/nvim/search.c:3242:  for should always use braces  [readability/braces] [5]
src/nvim/search.c:3257:  /*-style comment found, it should be replaced with //-style.  /*-style comments are only allowed inside macros.  Note that you should not use /*-style comments to document macros itself, use doxygen-style comments for this.  [readability/old_style_comment] [5]
src/nvim/search.c:3257:  if should always use braces  [readability/braces] [5]
src/nvim/search.c:3268:  if should always use braces  [readability/braces] [5]
src/nvim/search.c:3374:  for should always use braces  [readability/braces] [5]
src/nvim/search.c:3375:  Line contains only semicolon. If this should be an empty statement, use {} instead.  [whitespace/semicolon] [5]
src/nvim/spell.c:709:  for should always use braces  [readability/braces] [5]
src/nvim/spell.c:817:  for should always use braces  [readability/braces] [5]
src/nvim/spell.c:819:  If an else has a brace on one side, it should have it on both  [readability/braces] [5]
src/nvim/spell.c:832:  if should always use braces  [readability/braces] [5]
src/nvim/spell.c:891:  for should always use braces  [readability/braces] [5]
src/nvim/spell.c:1295:  if should always use braces  [readability/braces] [5]
src/nvim/spell.c:2315:  for should always use braces  [readability/braces] [5]
src/nvim/spell.c:2316:  if should always use braces  [readability/braces] [5]
src/nvim/spell.c:2327:  for should always use braces  [readability/braces] [5]
src/nvim/spell.c:2783:  while should always use braces  [readability/braces] [5]
src/nvim/spell.c:2786:  while should always use braces  [readability/braces] [5]
src/nvim/spell.c:2986:  if should always use braces  [readability/braces] [5]
src/nvim/spell.c:3954:  if should always use braces  [readability/braces] [5]
src/nvim/spell.c:4305:  if should always use braces  [readability/braces] [5]
src/nvim/spell.c:4897:  for should always use braces  [readability/braces] [5]
src/nvim/spell.c:4898:  Do not use preincrement in statements, use postincrement instead  [readability/increment] [5]
src/nvim/spell.c:4899:  for should always use braces  [readability/braces] [5]
src/nvim/spell.c:4900:  Do not use preincrement in statements, use postincrement instead  [readability/increment] [5]
src/nvim/spell.c:7520:  while should always use braces  [readability/braces] [5]
src/nvim/spell.c:7543:  if should always use braces  [readability/braces] [5]
src/nvim/spell.c:7551:  if should always use braces  [readability/braces] [5]
src/nvim/spellfile.c:1511:  if should always use braces  [readability/braces] [5]
src/nvim/spellfile.c:2531:  for should always use braces  [readability/braces] [5]
src/nvim/spellfile.c:2532:  if should always use braces  [readability/braces] [5]
src/nvim/spellfile.c:2534:  for should always use braces  [readability/braces] [5]
src/nvim/spellfile.c:2535:  if should always use braces  [readability/braces] [5]
src/nvim/spellfile.c:3074:  if should always use braces  [readability/braces] [5]
src/nvim/spellfile.c:3355:  for should always use braces  [readability/braces] [5]
src/nvim/spellfile.c:3355:  Do not use preincrement in statements, including for(;; action)  [readability/increment] [4]
src/nvim/spellfile.c:3357:  If an else has a brace on one side, it should have it on both  [readability/braces] [5]
src/nvim/spellfile.c:3368:  for should always use braces  [readability/braces] [5]
src/nvim/spellfile.c:3368:  Do not use preincrement in statements, including for(;; action)  [readability/increment] [4]
src/nvim/syntax.c:2827:  while should always use braces  [readability/braces] [5]
src/nvim/window.c:5104:  while should always use braces  [readability/braces] [5]
src/nvim/window.c:5106:  /*-style comment found, it should be replaced with //-style.  /*-style comments are only allowed inside macros.  Note that you should not use /*-style comments to document macros itself, use doxygen-style comments for this.  [readability/old_style_comment] [5]
src/nvim/charset.h:46:  Should leave no spaces after a cast: u'(char_u) c'  [whitespace/cast] [2]
```

janlazo commented:
I tried inline already but these macros receive `char` and `char_u` pointers so I get precision errors on compilation.

ckelsel commented:
Let's mark the macro-case changes as NA for now. see #8282

janlazo committed:
vim-patch:8.0.0451: some macros are in lower case

Problem:    Some macros are in lower case.
Solution:   Make a few more macros upper case. Avoid lower case macros use an
            argument twice.
https://github.com/vim/vim/commit/91acfffc1e6c0d8c2abfb186a0e79a5bf19c3f3f

justinmk commented on src/nvim/charset.h:
```diff
@@ -34,4 +34,15 @@ typedef enum {
+/// Used very often if 'linebreak' is set
+static inline bool vim_isbreak(int c)
+{
+  return breakat_flags[(char_u)c];
```
it's failing the "single include" check:

```
In file included from src/nvim/auto/charset.h.test-include.c:1:0:
/home/travis/build/neovim/neovim/src/nvim/charset.h: In function â€˜vim_isbreakâ€™:
/home/travis/build/neovim/neovim/src/nvim/charset.h:46:10: error: â€˜breakat_flagsâ€™ undeclared (first use in this function)
   return breakat_flags[(char_u)c];
          ^
/home/travis/build/neovim/neovim/src/nvim/charset.h:46:10: note: each undeclared identifier is reported only once for each function it appears in
```

I think `breakat_flags` could be moved to `option_defs.h`, then `#include "nvim/option_defs.h` from charset.h

janlazo committed:
charset: include option_defs.h for breakat_flags

justinmk commented:
nicely done!
